types/fbt/index.d.ts

```diff
@@ -1,171 +1,285 @@
+    getTranslatedInput: (input: unknown) => unknown;
+    getRegisteredTranslations: () => TranslationDict;
+    registerTranslations: (translations: TranslationDict) => void;
+    mergeTranslations: (newTranslations: TranslationDict) => void;
+};
 
-    type NameProps = Omit<ParamOptions, 'gender'> & {
-        name: string;
-        gender: IntlVariationsGenderValues;
-        children: React.ReactNode;
+export class FbtResult {
+    constructor(contents: unknown[], errorListener?: unknown);
+    getContents(): unknown;
+    // This relies on toString() which contains i18n logging logic to track impressions.
+    // I.e. If you use this, i18n will register the string as displayed!
+    toJSON(): string;
+    // Hack for allowing FBTResult to play nice in React components
+    _store?: {
+        validated: boolean;
     };
+}
 
-    interface SameNameProps {
-        name: string;
-    }
+// https://github.com/facebook/fbt/blob/d14af3e7dcf775562d4c4d27e62861bd9394ea6b/packages/babel-plugin-fbt/src/FbtConstants.js#L70-L75
+export type PronounType = 'object' | 'possessive' | 'reflexive' | 'subject';
 
-    interface FbtResult {
-        toString: () => string;
-    }
+// https://github.com/facebook/fbt/blob/d14af3e7dcf775562d4c4d27e62861bd9394ea6b/runtime/nonfb/GenderConst.js#L17-L32
+export enum GenderConst {
+    NOT_A_PERSON = 0,
+    FEMALE_SINGULAR = 1,
+    MALE_SINGULAR = 2,
+    FEMALE_SINGULAR_GUESS = 3,
+    MALE_SINGULAR_GUESS = 4,
+    MIXED_UNKNOWN = 5,
+    NEUTER_SINGULAR = 6,
+    UNKNOWN_SINGULAR = 7,
+    FEMALE_PLURAL = 8,
+    MALE_PLURAL = 9,
+    NEUTER_PLURAL = 10,
+    UNKNOWN_PLURAL = 11,
 }
 
-export const GenderConst: FBT.GenderConst;
-export const IntlVariations: FBT.IntlVariations;
-export const IntlViewerContext: FBT.IntlViewerContext;
+// https://github.com/facebook/fbt/blob/d14af3e7dcf775562d4c4d27e62861bd9394ea6b/runtime/nonfb/IntlVariations.js#L17-L31
+export enum IntlVariations {
+    BITMASK_NUMBER = 28,
+    BITMASK_GENDER = 3,
+    NUMBER_ZERO = 16,
+    NUMBER_ONE = 4,
+    NUMBER_TWO = 8,
+    NUMBER_FEW = 20,
+    NUMBER_MANY = 12,
+    NUMBER_OTHER = 24,
+    GENDER_MALE = 1,
+    GENDER_FEMALE = 2,
+    GENDER_UNKNOWN = 3,
+}
 
-// The Fbt* exports aren't real and is only used for syntax sugar.
-// https://github.com/facebook/fbt/blob/8607c1f2798ef18c6142a2cf1c5a9351c6d7df69/transform/babel-plugin-fbt/FbtUtil.js#L28-L40
-export const Fbt: React.FC<FBT.Props>;
-export const FbtEnum: React.FC<FBT.EnumProps>;
-export const FbtParam: React.FC<FBT.ParamProps>;
-export const FbtPlural: React.FC<FBT.PluralProps>;
-export const FbtPronoun: React.FC<FBT.PronounProps>;
-export const FbtName: React.FC<FBT.NameProps>;
-export const FbtSameParam: React.FC<FBT.SameNameProps>;
+export type IntlVariationsGender =
+    | IntlVariations.GENDER_MALE
+    | IntlVariations.GENDER_FEMALE
+    | IntlVariations.GENDER_UNKNOWN;
 
-export interface Fbt {
-    (source: string, desc: string, options?: FBT.Options): FBT.FbtResult;
```

Comment:
Here's the previous type definition, which is
```javascript
interface FbtResult {
    toString: () => string;
}
````

Reply:
@Swahvay because  `fbt`  returns array of string:

```tsx
fbt(`My name ${fbt.param('name', person.getName())}`) // ['Мое имя', person.getName()]
```

Reply:
How do you use `fbt` as a function in code then? Do you always have to write `fbt(…).join(‘’)`?

Reply:
Did some digging and fbt() returns an `FBTResult` object, which has a `toString()` method defined here: https://github.com/facebook/fbt/blob/main/runtime/shared/FbtResultBase.js#L90, which does return just a string.

Reply:
Sorry, I must have had some transforms mixed up. I was getting that type definition but after some reworking it's disappeared and everything is working correctly. Sorry for the confusion.

Reply:
Actually I'm sorry again. It's back and I'm not sure if there error is on my end or not anymore. I do know that `fbt()` should not return a `string[]` though.

Reply:
@Swahvay so what you propose to do?